fix NAT traversal for Nodes behind NAT by exposing coturn on 3478

rajpratik71 opened the pull request:
Before exposing port of coturn on 3478 For Nodes behind the NAT
are not able to ping other nodes and vice-versa as they depend and connect
coturn on Public_IP:3478


Signed-off-by: Pratik raj <rajpratik71@gmail.com>

renatof reviewed (changes requested):
Thanks for the request. Now that I look at it again, it's recommended by the coturn Docker container devs to have it bind to --network=host, because it uses a lot of ports - it uses not only port 3478 to listen to TURN requests, but also a range of ports (specified in turnserver.conf, by default 49160 - 59200 in the shell script) from which TURN dynamically allocates endpoints.
I'm not too familiar with docker-compose - is it possible to configure the yaml file to bind to --network=host?

rajpratik71 commented:
![image](https://user-images.githubusercontent.com/12658912/110023929-c07b5d80-7d53-11eb-8861-f10b20f94dc8.png)


from the deployment only '3478' is exposed by coturn docker container and also same is mentioned in docs at https://edgevpn.io/AWSbootstrap/

renatof commented:
Thanks for following up.

There is an extra rule in the AWS policies in https://edgevpn.io/AWSbootstrap/ for those ports:

Custom UDP Rule | UDP | 49160 - 59200 | 0.0.0.0/0 | coturn

The 49160 - 59200 range matches what is configured in turnserver.conf, from the shell script:

echo "min-port=49160" >> turnserver.conf
echo "max-port=59200" >> turnserver.conf

coturn needs at least port 3478 open as a well-known port to serve as a STUN server; to serve as a TURN server, it also needs to be able to receive requests through port 3478, and open additional TURN UDP endpoints on ports in this min-max range (one per TURN session). If you only leave port 3478 open, coturn will serve only as a STUN server, and will not work properly as a TURN server.

As per the coturn Docker documentation:

https://hub.docker.com/r/instrumentisto/coturn

While it's possible to map port 3478 and a the additional port range, "Or just use the host network directly (recommended, as Docker performs badly with large port ranges):

docker run -d --network=host instrumentisto/coturn
"

Ideally, we should be able to have the user configure min/max ports (and not hardwire them), and propagate those in the configuration of coturn (as well as matching rules in the AWS security group)

rajpratik71 committed:
fix NAT traversal for Nodes behind NAT by exposing coturn on 3478

Before exposing port of coturn on 3478 For Nodes behind the NAT
are not able to ping other nodes and vice-versa as they depend and connect
coturn on Public_IP:3478

Signed-off-by: Pratik raj <rajpratik71@gmail.com>

rajpratik71 commented:
@renatof  updated deployment to support host network with required ports

is now as asked and required
